### PR TITLE
fix: Add script to update MongoDB and Redis supervisor config when changing the URI

### PR DESCRIPTION
**Pull request by geekup-legodevops**
## Description

- The script will check for APPSMITH_MONGODB_URI and APPSMITH_REDIS_URL for **internal/external** and **enable/disable** supervisor config accordingly.
- Restart the related services including `backend, editor, rts`
Flow

Fixes # (issue)
- [[Bug]: Embedded MongoDB not removed immediately when changing the MongoDB URI](https://github.com/appsmithorg/appsmith/issues/9967)

## Type of change

- Bug fix (non-breaking change which fixes an issue)

## How Has This Been Tested?
- Have appsmith-ce running
- Update `APPSMITH_MONGODB_URI` ( using external src)
- Call `supervisor-mongodb.sh` script
- Check supervisord status.
## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: deploy/docker/scripts/supervisor-mongodb.sh [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+update_supervisord_mongodb_conf() {
+  echo "Update supervisord mongodb conf"
+  cd "$SUPERVISOR_CONF_PATH"
+  if [ $isUriLocal -eq 1 ]; then
+    echo "disable MongoDB supervisord"
+    ls | grep -i mongodb | sed -e "p;s/\.conf/\.disable/" | xargs -n2 mv
+  else
+    echo "enable MongoDB supervisord"
+    ls | grep -i mongodb | sed -e "p;s/\.disable/\.conf/" | xargs -n2 mv
```
Comment (sharat87) [inline deploy/docker/scripts/supervisor-mongodb.sh:29]: If local MongoDB was disabled originally, and then enabled, there would be no file called `mongodb.conf.disabled`. Please refer to the entrypoint script. If an external MongoDB is configured, we don't copy the `mongodb.conf` _at all_.

Can we instead remove the file when disabled, and copy it from the original source when it's enabled? Just like in the entrypoint script please?

File: deploy/docker/scripts/supervisor-mongodb.sh [hunk 2]
```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env bash
+
+set -e
```
Comment (sharat87) [inline deploy/docker/scripts/supervisor-mongodb.sh:3]: Let's enable xtrace so we get logs of what's happening when this script is run.

```suggestion
set -e
set -o xtrace
```

File: deploy/docker/scripts/supervisor-mongodb.sh [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+  else
+    echo "enable MongoDB supervisord"
+    ls | grep -i mongodb | sed -e "p;s/\.disable/\.conf/" | xargs -n2 mv
+  fi
+}
+
+check_mongodb_uri
+update_supervisord_mongodb_conf
+supervisorctl update
+supervisorctl restart backend editor rts
```
Comment (sharat87) [inline deploy/docker/scripts/supervisor-mongodb.sh:36]: The update command will immediately terminate the MongoDB process, leaving the backend server and RTS starting to throw errors. Can we restart those first, so they can connect to the external MongoDB, and then do a `supervisorctl update`?

```suggestion
supervisorctl restart backend editor rts
supervisorctl update
```

File: deploy/docker/scripts/supervisor-mongodb.sh [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+    echo "disable MongoDB supervisord"
+    ls | grep -i mongodb | sed -e "p;s/\.conf/\.disable/" | xargs -n2 mv
+  else
+    echo "enable MongoDB supervisord"
+    ls | grep -i mongodb | sed -e "p;s/\.disable/\.conf/" | xargs -n2 mv
+  fi
+}
+
+check_mongodb_uri
+update_supervisord_mongodb_conf
```
Comment (sharat87) [inline deploy/docker/scripts/supervisor-mongodb.sh:34]: Can we also fix the same problem for Redis as well please?

**Review by sharat87 — CHANGES_REQUESTED** [review]
Please rename the script to something that illustrates the purpose of the script. Something like `update-supervisor-and-restart.sh`.

File: deploy/docker/scripts/supervisor-mongodb.sh (hunk 2 above)
Comment (khoahuynhdev) [reply deploy/docker/scripts/supervisor-mongodb.sh]: i have updated this.

File: deploy/docker/scripts/supervisor-mongodb.sh (hunk 4 above)
Comment (khoahuynhdev) [reply deploy/docker/scripts/supervisor-mongodb.sh]: We do the check for local redis and update supervisor config as well

File: deploy/docker/scripts/supervisor-mongodb.sh (hunk 3 above)
Comment (khoahuynhdev) [reply deploy/docker/scripts/supervisor-mongodb.sh]: i have updated this.

File: deploy/docker/scripts/supervisor-mongodb.sh (hunk 1 above)
Comment (geekup-legodevops) [reply deploy/docker/scripts/supervisor-mongodb.sh]: we have updated following the logic in `entrypoint` file

File: deploy/docker/scripts/supervisor-mongodb.sh (hunk 4 above)
Comment (geekup-legodevops) [reply deploy/docker/scripts/supervisor-mongodb.sh]: we have updated logic to check and update for redis as well

**Review by sharat87 — APPROVED** [review]

**Pull request merged**